api/src/services/critterbase-service.ts

```diff
@@ -0,0 +1,249 @@
+  ['marking-materials']: `${lookups}/marking-materials`,
+  ['marking-types']: `${lookups}/marking-types`,
+  ['collection-categories']: `${lookups}/collection-unit-categories`,
+  taxons: lookupsTaxons,
+  species: `${lookupsTaxons}/species`,
+  colours: `${lookups}/colours`,
+
+  // lookups/enum
+  sex: `${lookupsEnum}/sex`,
+  ['critter-status']: `${lookupsEnum}/critter-status`,
+  ['cause-of-death-confidence']: `${lookupsEnum}/cod-confidence`,
+  ['coordinate-uncertainty-unit']: `${lookupsEnum}/coordinate-uncertainty-unit`,
+  ['frequency-units']: `${lookupsEnum}/frequency-units`,
+  ['measurement-units']: `${lookupsEnum}/measurement-units`,
+
+  // xref
+  ['collection-units']: `${xref}/collection-units`,
+
+  // taxon xrefs
+  ['taxon-measurements']: `${xref}/taxon-measurements`,
+  ['taxon_qualitative_measurements']: `${xref}/taxon-qualitative-measurements`,
+  ['taxon-qualitative-measurement-options']: `${xref}/taxon-qualitative-measurement-options`,
+  ['taxon-quantitative-measurements']: `${xref}/taxon-quantitative-measurements`,
+  ['taxon-collection-categories']: `${xref}/taxon-collection-categories`,
+  ['taxon-marking-body-locations']: `${xref}/taxon-marking-body-locations`
+} as const;
+
+export type ICbRouteKey = keyof typeof CbRoutes;
+
+export const CRITTERBASE_API_HOST = process.env.CB_API_HOST || ``;
+const CRITTER_ENDPOINT = '/critters/';
+const BULK_ENDPOINT = '/bulk';
+const SIGNUP_ENDPOINT = '/signup';
+const FAMILY_ENDPOINT = '/family';
+
+export class CritterbaseService {
+  user: ICritterbaseUser;
+  keycloak: KeycloakService;
+
+  constructor(user: ICritterbaseUser) {
+    this.user = user;
+    this.keycloak = new KeycloakService();
+  }
+
+  async getToken(): Promise<string> {
+    const token = await this.keycloak.getKeycloakServiceToken();
+    return token;
+  }
+
+  getUserHeader(): { user: string } {
+    return { user: JSON.stringify(this.user) };
+  }
+
+  async makeGetRequest(endpoint: string, params: QueryParam[]) {
+    let url = `${CRITTERBASE_API_HOST}${endpoint}`;
+    for (let i = 0; i < params.length; i++) {
+      url += (i === 0 ? '?' : '&') + params[i].key + '=' + params[i].value;
+    }
+    console.log('Actual URL we will hit ' + url);
+    const token = await this.getToken();
+    const response = await axios.get(url, {
+      headers: {
+        authorization: `Bearer ${token}`,
+        ...this.getUserHeader()
+      }
+    });
+    return response.data;
+  }
+
+  async makePostPatchRequest(method: 'post' | 'patch', endpoint: string, data?: any) {
+    const url = `${CRITTERBASE_API_HOST}${endpoint}`;
+    const token = await this.getToken();
+    const response = await axios[method](url, data, {
+      headers: {
+        authorization: `Bearer ${token}`,
+        ...this.getUserHeader()
+      }
+    });
+    return response.data;
+  }
+
+  async getLookupValues(route: ICbRouteKey, params: QueryParam[]) {
+    return this.makeGetRequest(`${CbRoutes[route]}`, params);
```

Comment:
Nit: if you want to cast this as a String, use `String(CbRoutes[route])` instead of a template string